### ISSUE TITLE
Cannot construct a Request with a Request object that has already been used

### ISSUE THREAD

**Issue by radum** [DataDog/dd-trace-js#4259]
Hello,

```
Next.js: 14.2.2
DDTrace: 5.11.0
Node: 20.11.1
Running on AWS EKS pods
```

We just upgraded to version 5.11.0 and our Next.js internal API routes that are using fetch to get data from external services are now failing with `Cannot construct a Request with a Request object that has already been used.`. Disabling datadog or reverting to the previous version (5.10.0) works fine.

I will try to see if I can setup a demo repo but its very unlikely I can strip down our setup.

We load DD via `"start": "node -r ./server-preload.js ./node_modules/.bin/next start"` and in that file we have:

```js
function setUpDatadogTracing() {
	const tracer = require('dd-trace');

	tracer.init({
		runtimeMetrics: true,
		logInjection: true,
		// env: 'dev', 		// This is set on the POD automatically via DATADOG_ENV=
		// service: '',		// This is set on the POD automatically via DD_SERVICE=
		// version: packageJSON?.version ?? 'unknown'		// This is set on the POD automatically via DD_VERSION=
	});
	tracer.use('next');
}

setUpDatadogTracing();
```

Not sure what changed but looks like fetch constructor is called twice from DD and then within our app it fails to construct it again.

**Comment by cannontrodder** [comment]
We have seen this on the 4.x.x release line, unfortunately we needed this to get a fix related to an out of memory race condition on the auto injection docker image

**Comment by yjukaku** [comment]
We are also seeing the same issue on 4.x, and Next.js 13.x

`fetch` fails with an error when making a POST with a body to our backend. We think it fails only when the server responds with a redirect (ours is using a 307).

**Comment by radum** [comment]
@yjukaku That is very interesting, have you found an workaround for it?

**Comment by Klaitos** [comment]
Same issue here,  lost few hours to land on this issue.

```
TypeError: Cannot construct a Request with a Request object that has already been used.
at node:internal/deps/undici/undici:12618:11
```

**Comment by cannontrodder** [comment]
Looks like this commit is designed to fix it: https://github.com/DataDog/dd-trace-js/commit/0f36e48775b1fe47ee4de18eeffae6cf3ac761aa

**Comment by yjukaku** [comment]
Our workaround, other than downgrading the DD library, was to update the URLs we were fetching to avoid the redirect.

**Comment by mdnorman** [comment]
We ran into this as well. Even though we're pinning the versions of [datadog-lambda-extension](https://github.com/DataDog/datadog-lambda-extension/releases) and [datadog-lambda-js](https://github.com/Datadog/datadog-lambda-js/releases), the Lambda layer was still picking up a newer version of dd-trace that had this problem.

**Comment by tbauer428** [comment]
Ran into this same error with `next-auth` using dd-trace in our Dockerfile, we had `dd-trace@^5`, which pulled in `v5.11.0`.

Downgrading to `v5.9.0` immediately alleviated the issue.

**Comment by juancarlosjr97** [comment]
Same issue here, and we are pinning to v5.10.0 for now.

**Comment by olafurw** [comment]
Seems a fix has existed for a couple of days but hasn't been pushed out yet #4258

**Comment by KevinVR** [comment]
Can confirm - issue on 5.11.0, while 5.10.0 works fine. 

We need PR https://github.com/DataDog/dd-trace-js/pull/4258 also for v5, not only pre-v6.

**Comment by EmiiFont** [comment]
Ran into this same error with auth0 node sdk package using dd-trace 

Downgrading to v5.10.0 immediately alleviated the issue.

**Comment by rkstamm** [comment]
Same here. Caused our next.js server side fetches to fail all over. Storing a local copy of the wrapper using specific versioning to 5.10, and from now on.

**Comment by oachoor** [comment]
can confirm as well (nestjs), downgrading to v5.10.0 for now

**Comment by anddreluis2** [comment]
if you remove the connection to apm the app works again

**Comment by tlhunter** [comment]
A fix for this has been released in v5.12.0, v4.36.0, and v3.57.0 of the library. Please upgrade to confirm that it fixes the issue. If it does not fix the issue please @ me and I'll reopen this issue and we'll investigate further. Thanks!

**Comment by juancarlosjr97** [comment]
Nice! Looks like it is working :)
